### PR TITLE
Validation test model

### PR DESCRIPTION
**Pull request by AhmadiGu**
### Validation test model branch
*The following changes introduced:* 
**Important:** Kindly check this commit I forgot to create the branch first https://github.com/AhmadiGu/Bolg_App_Rails/pull/3/commits/e658c741b923ff107aec5d0ec19f82fb68101d10

- write validation for the user and post models 
- write the test for the user model  
- write the test for the post model  
- write the test for the comment model  
- write the test for the like model

### CODE REVIEW

File: spec/models/user_spec.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+  context 'Testing user behavior' do
+    user = User.new(name: 'Ahmad')
+    user.id = 1
+    user.save
+
+    before { 10.times { Post.create(author_id: user.id, title: 'Title', text: 'Text') } }
+    it 'List the most 5 recent posts' do
+      expect(user.recent_posts.length).to eq 5
+    end
```
Comment (pinkmoon25) [inline spec/models/user_spec.rb]: - [ ] Good job adding validations and testing your model, Just this one failing tests you need to fix :point_down: 
image

What's happening here is your user is not being saved and also the post is missing comments_counter and likes_counter and is also not being saved, a better way would be to pass `author: subject` while creating the posts, one more thing id your recent posts method is limiting to 3 results so you expecting 5 here will not pass.

**Review by pinkmoon25 — CHANGES_REQUESTED** [review]
Hi @AhmadiGu , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
rubocop :heavy_check_mark: 
gitflow :heavy_check_mark: 
Validations :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/models/user_spec.rb [hunk 2]
```diff
@@ -22,14 +22,19 @@
...
   end
 
+  after { subject.save }
   context 'Testing user behavior' do
-    user = User.new(name: 'Ahmad')
-    user.id = 1
-    user.save
+    # user = User.new(name: 'Ahmad')
+    # user.id = 1
+    # user.save
```
Comment (fobadara) [inline spec/models/user_spec.rb]: [REQUIRED] Kindly remove unused code. This makes your project look more professional

**Review by fobadara — CHANGES_REQUESTED** [review]
Hi @AhmadiGu, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which haven't been addressed yet.


 ## Highlights
✅ Linters are passing

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by MeqdamAlqudah — APPROVED** [review]
Hi @AhmadiGu 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
image

## Highlights 
:+1: no linter errors 
:+1: files structure

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**